src/gpu_hash_table.cuh

```diff
@@ -134,7 +134,35 @@ class gpu_hash_table {
     cudaEventDestroy(stop);
     return temp_time;
   }
+  float hash_build(KeyT* h_key, ValueT* h_value, uint32_t num_keys, bool unique_keys) {
```

Comment:
make this also a separate function, rather than an overloaded function with different argument. Are you doing this because the cset class doesn't support it? If so, just please make another function for cset version, and then just put some assert into it to indicate we haven't yet implemented the feature for that path. so the function should compile, but no one can use it yet.